### PR TITLE
4 complete accountdefinitionapi implementation

### PR DESCRIPTION
**Pull request by karolkadlubowski**
PR flp nie hejtuj

### CODE REVIEW

File: src/Services/AccountDefinition.API/tests/AccountDefinition.API.Tests.Unit/Application/Services/AccountProviderServiceTests.cs [hunk 1]
```diff
@@ -98,10 +104,44 @@ public async Task AddAccountProviderAsync_WhenAccountProviderAlreadyExistsInData
...
+            //Assert
+            result.Should().Be(ProviderId);
+        }
+
+        [Test]
+        public async Task DeleteAccountProviderAsync_WhenAccountProviderDoesNotExistInDatabase_ThrowEntityNotFoundException()
+        {
+            //Arrange
+            _accountProviderRepository.Setup(x => x.DeleteAccountProviderByIdAsync(ProviderId))
+                .ReturnsAsync(0);
```
Comment (TomeQ2k) [inline src/Services/AccountDefinition.API/tests/AccountDefinition.API.Tests.Unit/Application/Services/AccountProviderServiceTests.cs:136]: You can change 0 to default

File: src/Services/AccountDefinition.API/src/AccountDefinition.API.Application/Services/AccountProviderService.cs [hunk 2]
```diff
@@ -29,6 +30,17 @@ public class AccountProviderService : IAccountProviderService
             _logger = logger;
         }
 
+        public async Task<IReadOnlyList<AccountProviderDto>> GetAccountProvidersAsync()
+        {
+            var accountProviderPersistenceModels = await _accountProviderRepository.GetAccountProvidersAsync();
+
+            var accountProviders = _mapper.Map<IReadOnlyList<AccountProvider>>(accountProviderPersistenceModels);
+            
+            _logger.Info($"{accountProviders.Count} account types fetched from the database");
```
Comment (filippyziak) [inline src/Services/AccountDefinition.API/src/AccountDefinition.API.Application/Services/AccountProviderService.cs:39]:
```suggestion
            _logger.Info($"{accountProviders.Count} account providers fetched from the database");
```

File: src/Services/AccountDefinition.API/src/AccountDefinition.API/Controllers/AccountProviderController.cs [hunk 3]
```diff
@@ -21,6 +22,22 @@ public AccountProviderController(IMediator mediator, ILogger logger) : base(medi
...
+        public async Task<IActionResult> GetAccountProviders([FromQuery] GetAccountProvidersQuery query)
+        {
+            _logger.Info($"Sending query: {nameof(GetAccountProvidersQuery)}");
+
+            var response = await _mediator.Send(query);
+
+            return this.CreateResponse(response);
+        }
+        
+
```
Comment (filippyziak) [inline src/Services/AccountDefinition.API/src/AccountDefinition.API/Controllers/AccountProviderController.cs:40]: remove this line
```suggestion
```

**Review by filippyziak — CHANGES_REQUESTED** [review]
as in comments

**Review by TomeQ2k — APPROVED** [review]

**Pull request merged**